R/pipe_continuation_linter.R

```diff
@@ -3,14 +3,56 @@
 #' Check that each step in a pipeline is on a new line, or the entire pipe fits on one line.
 #'
 #' @evalRd rd_tags("pipe_continuation_linter")
+#'
+#' @examples
+#' # will produce lints
+#' code_lines <- "1:3 %>%\n mean() %>% as.character()"
+#' cat(code_lines)
+#' lint(
+#'   text = code_lines,
+#'   linters = pipe_continuation_linter()
+#' )
+#'
+#' code_lines <- "1:3 |> mean() |>\n     as.character()"
```

Comment:
is this example a mistake? it won't lint right?

Reply:
oh NVM, I am in the wrong linter in my head :)

Reply:
It does lint:

``` r
library(lintr)

code_lines <- "1:3 |> mean() |>\n as.character()"
writeLines(code_lines)
#> 1:3 |> mean() |>
#>  as.character()
lint(
  text = code_lines,
  linters = pipe_continuation_linter()
)
#> <text>:1:15: style: [pipe_continuation_linter] `|>` should always have a space before it and a new line after it, unless the full pipeline fits on one line.
#> 1:3 |> mean() |>
#>               ^~
```

<sup>Created on 2022-10-10 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>